includes/polling/bgp-peers.inc.php

```diff
@@ -178,14 +183,24 @@
             );
             data_update($device, 'bgp', $tags, $fields);
 
-            $peer['update']['bgpPeerState']              = $bgpPeerState;
-            $peer['update']['bgpPeerAdminStatus']        = $bgpPeerAdminStatus;
-            $peer['update']['bgpPeerFsmEstablishedTime'] = set_numeric($bgpPeerFsmEstablishedTime);
-            $peer['update']['bgpPeerInUpdates']          = set_numeric($bgpPeerInUpdates);
-            $peer['update']['bgpLocalAddr']              = $bgpLocalAddr;
-            $peer['update']['bgpPeerOutUpdates']         = set_numeric($bgpPeerOutUpdates);
+            $bgpPeers_fields = array(
+                'bgpPeerState'              => $bgpPeerState,
+                'bgpPeerAdminStatus'        => $bgpPeerAdminStatus,
+                'bgpPeerFsmEstablishedTime' => $bgpPeerFsmEstablishedTime,
+                'bgpPeerInUpdates'          => $bgpPeerInUpdates,
+                'bgpLocalAddr'              => $bgpLocalAddr,
+                'bgpPeerOutUpdates'         => $bgpPeerOutUpdates,
+            );
 
-            dbUpdate($peer['update'], 'bgpPeers', '`device_id` = ? AND `bgpPeerIdentifier` = ?', array($device['device_id'], $peer['bgpPeerIdentifier']));
+            foreach ($bgpPeers_fields as $db_field => $value) {
```

Comment:
I think you could actually use array_diff here.

It worked out pretty nicely for my entity state sensors.

```
$peer['update] = array_diff($bgpPeers_fields, $peer);
```
Will only return the fields from bgpPeers_fields that don't match one in $peer.

Reply:
Good shout, updated.